### PR TITLE
fix: updating last messages depending on last message update

### PR DESCRIPTION
**Pull request by vanbasten17**
_Remember to tag the PR with **WIP** tag if it's not ready to be merged_.  
[Reference](https://blog.codeminer42.com/on-writing-a-great-pull-request-37c60ce6f31d/)

## Description
* Added methods with @guinii  to set a `lastMessageUpdate` in order to be read after and ensure all the messages are loaded in the enduser side in the right way.

## Context
If the agent was the last to write in a chat, the information of some messages was missing if the enduser closed the tab of Botonic.

## Approach taken / Explain the design

Store and keep updaing a lastMessageUpdate timestamp to keep tracking messages to be updated correctly.

## Testing

The pull request...

- [ ] has unit tests
- [ ] has integration tests
- [X] doesn't need tests because... It's quite critical

### CODE REVIEW

File: packages/botonic-core/src/hubtype-service.js [hunk 1]
```diff
@@ -5,17 +5,19 @@ const PUSHER_KEY = process.env.WEBCHAT_PUSHER_KEY || '434ca667c8e6cb3f641c'
...
 
 export class HubtypeService {
-  constructor({ appId, user, lastMessageId, onEvent }) {
+  constructor({ appId, user, lastMessageId, lastMessageUpdateDate, onEvent }) {
     this.appId = appId
     this.user = user || {}
     this.lastMessageId = lastMessageId
+    this.lastMessageUpdateDate = lastMessageUpdateDate
     this.onEvent = onEvent
     if (user.id && lastMessageId) this.init()
```
Comment (ericmarcos) [inline packages/botonic-core/src/hubtype-service.js:14]:
```
if (user.id && (lastMessageId || lastMessageUpdateDate)) this.init()
```

File: packages/botonic-react/src/webchat/webchat-reducer.js [hunk 2]
```diff
@@ -39,6 +39,11 @@ export function webchatReducer(state, action) {
         messagesJSON: [],
         messagesComponents: [],
       }
+    case 'updateLastMessageDate':
+      return {
+        ...state,
+        lastMessageUpdate: action.payload || undefined,
```
Comment (ericmarcos) [inline packages/botonic-react/src/webchat/webchat-reducer.js:45]: this "||" is kind of redundant no?

File: packages/botonic-core/src/hubtype-service.js (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-core/src/hubtype-service.js]: good catch!

File: packages/botonic-react/src/webchat/webchat-reducer.js (hunk 2 above)
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/webchat-reducer.js]: yes, fixed 👍

**Pull request merged**